### PR TITLE
[Search] Add ResetDocs and ResetSkills rest client operations

### PR DESCRIPTION
**Pull request by Mohit-Chakraborty**
Update client type names that have been changed in REST spec.

### CODE REVIEW

**Comment by Mohit-Chakraborty** [comment]
cc: @tg-msft

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 1]
```diff
@@ -807,6 +819,17 @@ public partial class ImageAnalysisSkill : Azure.Search.Documents.Indexes.Models.
         public static bool operator !=(Azure.Search.Documents.Indexes.Models.ImageDetail left, Azure.Search.Documents.Indexes.Models.ImageDetail right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class IndexerCurrentState
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:822]: This should probably just be called `IndexerState` and then the property name can indicate the "current" nature of the result.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 2]
```diff
@@ -807,6 +819,17 @@ public partial class ImageAnalysisSkill : Azure.Search.Documents.Indexes.Models.
...
+    {
+        internal IndexerCurrentState() { }
+        public string AllDocsFinalChangeTrackingState { get { throw null; } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:825]: What does this property do?  What are the valid values?  Should there be an extensible enum with possibilities?  Could we think of a better name?  Ditto for the related properties.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 3]
```diff
@@ -807,6 +819,17 @@ public partial class ImageAnalysisSkill : Azure.Search.Documents.Indexes.Models.
...
+        public string AllDocsInitialChangeTrackingState { get { throw null; } }
+        public Azure.Search.Documents.Indexes.Models.IndexingMode? Mode { get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<string> ResetDatasourceDocumentIds { get { throw null; } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:828]: I think that should be `DataSource`

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 4]
```diff
@@ -837,6 +861,7 @@ public partial class IndexerExecutionResult
         public int ItemCount { get { throw null; } }
         public System.DateTimeOffset? StartTime { get { throw null; } }
         public Azure.Search.Documents.Indexes.Models.IndexerExecutionStatus Status { get { throw null; } }
+        public Azure.Search.Documents.Indexes.Models.IndexerExecutionStatusDetail? StatusDetail { get { throw null; } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:864]: What does this tell us?  How to interpret `Running`?  Can we think of a better name?

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:825]: Do we use `Docs` as an abbreviation anywhere else?

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 5]
```diff
@@ -846,12 +871,47 @@ public enum IndexerExecutionStatus
...
         Error = 1,
         Running = 2,
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct IndexingMode : System.IEquatable<Azure.Search.Documents.Indexes.Models.IndexingMode>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public IndexingMode(string value) { throw null; }
+        public static Azure.Search.Documents.Indexes.Models.IndexingMode IndexingAllDocs { get { throw null; } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:903]: Do we need to duplicate `Indexing` in this name (i.e., `IndexingMode.IndexingAllDocs` vs. `IndexingMode.AllDocs`).

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 6]
```diff
@@ -2513,6 +2573,12 @@ public partial class CaptionResult
         public string Highlights { get { throw null; } }
         public string Text { get { throw null; } }
     }
+    public partial class DocumentKeysOrIds
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2576]: This class name isn't super helpful.  Shouldn't this be something more like `ResetDocumentOptions`?

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 7]
```diff
@@ -2513,6 +2573,12 @@ public partial class CaptionResult
...
+    {
+        public DocumentKeysOrIds() { }
+        public System.Collections.Generic.IList<string> DatasourceDocumentIds { get { throw null; } }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2579]: `DataSource` casing

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 8]
```diff
@@ -2681,6 +2747,11 @@ public partial class RangeFacetResult<T> where T : struct
         public T? From { get { throw null; } }
         public T? To { get { throw null; } }
     }
+    public partial class ResettingSkills
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2750]: This doesn't read like a .NET class name

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 9]
```diff
@@ -294,8 +302,12 @@ public partial class SearchIndexerClient
         public virtual System.Threading.Tasks.Task<Azure.Response<System.Collections.Generic.IReadOnlyList<string>>> GetSkillsetNamesAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Response<System.Collections.Generic.IReadOnlyList<Azure.Search.Documents.Indexes.Models.SearchIndexerSkillset>> GetSkillsets(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<System.Collections.Generic.IReadOnlyList<Azure.Search.Documents.Indexes.Models.SearchIndexerSkillset>>> GetSkillsetsAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response ResetDocs(string indexerName, bool? overwrite = default(bool?), Azure.Search.Documents.Models.DocumentKeysOrIds keysOrIds = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:305]: Should probably be `ResetDocuments`

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 8 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2750]: This is a class with just one `IList` property in it. With the current name on the service-side (`skillNames`), it is hard to imagine more properties being added.
Anyways, I was thinking of other options - `SkillsToReset`, `SkillsBeingReset`.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 1 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:822]: There is no property talking about the 'current' state. The 'initial' and 'final' states are updated as appropriate.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 2 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs]: I have substituted instances of `Docs` with `Documents`.

Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs]: I am thinking we work with the service team to arrive at better names and update in the next preview.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 4 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:853]: I am thinking we work with the service team to arrive at better names and update in the next preview.

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs]: `IndexerExecutionResult` has a property of this type.  I assume that's meant to be "current"?

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs (hunk 2 above)
Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs]: I'll do that.

**Review by tg-msft — APPROVED** [review]
Looks good!

**Pull request merged**